packages/docregistry/src/registry.ts

```diff
@@ -960,6 +960,14 @@ export namespace DocumentRegistry {
      * Whether to activate the widget.  Defaults to `true`.
      */
     activate?: boolean;
+
+    /**
+     * The rank order of the widget among its siblings.
+     *
+     * #### Notes
+     * This field may be used or ignored depending on shell implementation.
+     */
+    rank?: number;
```

Comment:
I think my above objection to using `DocumentRegistry.IOpenOptions` might be partially due to the fact that it exists in the Document Registry, which is a little weird for such a shell-oriented set of options. What if we moved these to the `ILabShell` namespace?

Reply:
I think that it's fine for them to be in the `JupyterFrontEnd.Shell` namespace. What do we gain from them being in the `DocumentRegistry` at all?

Reply:
I don't know the full rationale for why they were there to begin with so I didn't want to cavalierly move them. If we decide to go with the above approach, I'll investigate it a bit more and if it truly is the case that none of it belongs in `DocumentRegistry` then I can make a single interface called `IAddOptions` or some such in `JupyterFrontEnd.Shell`. But if it looks like some of it belongs in the document registry, then I can create the new interface and inherit from the old one.

Reply:
It looks to me like the docregistry really only uses them in one place, in `Context`. Relatedly, I have been thinking for a while that `Context` might be better located in the `docmanager`.

Reply:
Hmm, on the other hand, including `IOpenOptions` in `@jupyterlab/application` would introduce new dependencies in several places, and I'm not sure I like that...

Reply:
I sorta like that this is "document opening options" that the shell has basically upgraded with some albeit shell-specific things, but it stays in the document registry.